com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/helpers/ClientHelperTest.java

```diff
@@ -0,0 +1,119 @@
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloud.lm.sl.cf.core.model.CloudTarget;
+import com.sap.cloud.lm.sl.cf.core.util.ApplicationURI;
+
+public class ClientHelperTest {
+
+    private static final String ORG_NAME = "deploy-service";
+    private static final String SPACE_NAME = "i123456";
+    private static final UUID GUID = UUID.fromString("040a3dd6-25bd-464d-8804-97a0d95dfdbd");
+    private static final String SPACE_ID = "8819b12c-6dde-4338-8530-93b2fba56df6";
+
+    @Mock
+    private CloudControllerClient client;
+
+    private ClientHelper clientHelper;
+
+    public ClientHelperTest() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @BeforeEach
+    public void setUp() {
+        clientHelper = new ClientHelper(client);
+    }
+
+    @Test
+    public void testDeleteRoute() {
+        String uri = "https://deploy-service.cfapps.sap.hana.ondemand.com";
+        clientHelper.deleteRoute(uri);
+        ApplicationURI route = new ApplicationURI(uri);
+        Mockito.verify(client)
+               .deleteRoute(route.getHost(), route.getDomain(), route.getPath());
+    }
+
+    @Test
+    public void testComputeSpaceId() {
+        Mockito.when(client.getSpace(ORG_NAME, SPACE_NAME, false))
+               .thenReturn(createCloudSpace(GUID, SPACE_NAME, ORG_NAME));
+        String spaceId = clientHelper.computeSpaceId(ORG_NAME, SPACE_NAME);
+        Assertions.assertEquals(GUID.toString(), spaceId);
+    }
+
+    @Test
+    public void testComputeSpaceIdIfSpaceIsNull() {
+        Assertions.assertNull(clientHelper.computeSpaceId(ORG_NAME, SPACE_NAME));
+    }
+
+    @Test
+    public void testComputeTarget() {
+        Mockito.when(client.getSpace(Matchers.any(UUID.class)))
+               .thenReturn(createCloudSpace(GUID, SPACE_NAME, ORG_NAME));
+        CloudTarget target = clientHelper.computeTarget(SPACE_ID);
+        Assertions.assertEquals(ORG_NAME, target.getOrg());
+        Assertions.assertEquals(SPACE_NAME, target.getSpace());
+    }
+
+    @Test
+    public void testComputeTargetCloudOperationExceptionForbiddenThrown() {
+        Mockito.when(client.getSpace(Matchers.any(UUID.class)))
+               .thenThrow(new CloudOperationException(HttpStatus.FORBIDDEN));
+        Assertions.assertNull(clientHelper.computeTarget(SPACE_ID));
+    }
+
+    @Test
+    public void testComputeTargetCloudOperationExceptionNotFoundThrown() {
+        Mockito.when(client.getSpace(Matchers.any(UUID.class)))
+               .thenThrow(new CloudOperationException(HttpStatus.NOT_FOUND));
+        Assertions.assertNull(clientHelper.computeTarget(SPACE_ID));
+    }
+
+    @Test
+    public void testComputeTargetCloudOperationExceptionBadRequestThrown() {
```

Comment:
The test asserts both the type of the exception and a parameter of that type.
Maybe split them into two tests?
- test...ThrowsCloudOpExc()
- test...ThrowsBadRequest()

Reply:
The test asserts BadRequest will be thrown if the client.getSpace throws BadRequest.